### ISSUE TITLE
False positive: def-fn in syntax-quote

### ISSUE THREAD

**Issue by NoahTheDuke** [NoahTheDuke/splint#26]
```
src/clojure/lazytest/core.clj:271:6 [style/def-fn] - Prefer `defn` instead of `def` wrapping `fn`.
(def ~(vary-meta test-name assoc :type :lazytest/var)
  (fn
   ~(gensym (str test-name "_"))
   []
   (binding [*context* nil] (assoc (describe ~@body) :type :lazytest/var))))
Consider using:
(defn ~(vary-meta test-name assoc :type :lazytest/var)
  ~(gensym (str test-name "_"))
  []
  (binding [*context* nil] (assoc (describe ~@body) :type :lazytest/var)))
```

**Comment by NoahTheDuke** [comment]
Fixed in 6440f384de51c6859432f5d429d780c0ebfef78b
